src/crypto/crypto_dh.cc

```diff
@@ -539,62 +539,68 @@ WebCryptoKeyExportStatus DHKeyExportTraits::DoExport(
 }
 
 namespace {
-AllocatedBuffer StatelessDiffieHellman(
-    Environment* env,
-    ManagedEVPPKey our_key,
-    ManagedEVPPKey their_key) {
-  size_t out_size;
-
-  EVPKeyCtxPointer ctx(EVP_PKEY_CTX_new(our_key.get(), nullptr));
+Maybe<bool> CreateStatelessDiffieHellmanContext(
+    const ManagedEVPPKey& our_key,
+    const ManagedEVPPKey& their_key,
+    char** buf,
+    size_t* out_size) {
+  const EVPKeyCtxPointer ctx(EVP_PKEY_CTX_new(our_key.get(), nullptr));
   if (!ctx ||
       EVP_PKEY_derive_init(ctx.get()) <= 0 ||
       EVP_PKEY_derive_set_peer(ctx.get(), their_key.get()) <= 0 ||
-      EVP_PKEY_derive(ctx.get(), nullptr, &out_size) <= 0)
-    return AllocatedBuffer();
+      EVP_PKEY_derive(ctx.get(), nullptr, out_size) <= 0)
+    return Nothing<bool>();
```

Comment:
There is no exception being thrown from this function, is there? Or am I missing something?

Reply:
@addaleax OpenSSL uses error codes to represent failures. Should we not treat those in the way we treat exceptions?

Reply:
@RaisinTen `Maybe<>` and `MaybeLocal<>` specifically communicate whether a JS exception is pending or not. I don’t think we should be using these for OpenSSL state.

Reply:
@addaleax ah, got it. Updated to use plain `bool`s, PTAL. :)